src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class IsoscelesTrapezoid extends Figure {
+    private final double topBase;
+    private final double bottomBase;
+    private final double side;
+
+    {
+        setName("isosceles trapezoid");
+    }
+
+    public IsoscelesTrapezoid(Color color, double topBase, double bottomBase, double side) {
+        super(color);
+        this.topBase = topBase;
+        this.bottomBase = bottomBase;
+        this.side = side;
+    }
+
+    public IsoscelesTrapezoid() {
+        super(new ColorSupplier().getRandomColor());
+        Random random = new Random();
+        this.topBase = random.nextInt(Figure.MAX_RANDOM_SIDE_VALUE);
+        this.bottomBase = random.nextInt(Figure.MAX_RANDOM_SIDE_VALUE);
+        this.side = random.nextInt(Figure.MAX_RANDOM_SIDE_VALUE);
```

Comment:
Let's move this code to FigureSupplier class. 
There are shouldn't be any business logic in the constructor.